ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala

```diff
@@ -6,129 +6,143 @@ package com.daml.platform.store.dao
-            KeyWithMaintainers(VersionedValue(ValueVersions.acceptedVersions.head, ValueParty(bob)), Set(bob))
-          ),
-        ),
-        parent = rootExercise,
-      )
-      val nestedExercise = builder.add(
-        NodeExercises(
-          targetCoid = create2,
-          templateId = someTemplateId,
-          choiceId = Ref.ChoiceName.assertFromString("SomeChoice"),
-          optLocation = None,
-          consuming = true,
-          actingParties = Set(bob),
-          chosenValue = VersionedValue(ValueVersions.acceptedVersions.head, someValueRecord),
-          stakeholders = Set(bob),
-          signatories = Set(bob),
-          children = ImmArray.empty,
-          exerciseResult = None,
-          key = Some(
-            KeyWithMaintainers(VersionedValue(ValueVersions.acceptedVersions.head, ValueParty(bob)), Set(bob))
-          ),
         ),
-        parent = rootExercise,
+        roots = ImmArray(nid)
       )
-      builder.add(
-        NodeCreate(
-          coid = toCid("cid3"),
-          coinst = someContractInstance,
-          optLocation = None,
-          signatories = Set(bob),
-          stakeholders = Set(alice, bob),
-          key = Some(
-            KeyWithMaintainers(VersionedValue(ValueVersions.acceptedVersions.head, ValueParty(bob)), Set(bob))
-          )
+      cid -> tx
+    }
+    val tx3: GenTransaction.WithTxValue[EventId, ContractId] = {
+      val nidRootExercise = EventId.assertFromString("#2:0")
+      val nidFetch = EventId.assertFromString("#2:1")
+      val nidNestedExercise = EventId.assertFromString("#2:2")
+      val nidCreate = EventId.assertFromString("#2:3")
+      val cid = ContractId.assertFromString("#contract3")
+      GenTransaction(
+        nodes = HashMap(
+          nidRootExercise ->
+            NodeExercises[EventId, ContractId, VersionedValue[ContractId]](
+              targetCoid = create1,
+              templateId = someTemplateId,
+              choiceId = Ref.ChoiceName.assertFromString("SomeChoice"),
+              optLocation = None,
+              consuming = true,
+              actingParties = Set(bob),
+              chosenValue = someVersionedCreateArg,
+              stakeholders = Set(alice, bob),
+              signatories = Set(alice),
+              children = ImmArray(nidFetch, nidNestedExercise),
+              exerciseResult = None,
+              key = None,
+            ),
+          nidFetch ->
+            NodeFetch[ContractId, VersionedValue[ContractId]](
+              coid = create2,
+              templateId = someTemplateId,
+              optLocation = None,
+              actingParties = Some(Set(bob)),
+              signatories = Set(bob),
+              stakeholders = Set(bob),
+              key = Some(
+                KeyWithMaintainers(
+                  VersionedValue(ValueVersions.acceptedVersions.head, ValueParty(bob)),
+                  Set(bob))
+              ),
+            ),
+          nidCreate ->
```

Comment:
I assume the order doesn’t matter here or is there a reason why you swapped `nidCreate` and `nestedExercise`?

Reply:
It's relevant for readability, pushing a fix.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6612/commits/c35fa284c8e3c5836a113b5da1cd34a84d795109